src/contrib/dependencyinjection/Akka.DependencyInjection/IDependencyResolver.cs

```diff
@@ -0,0 +1,21 @@
+// //-----------------------------------------------------------------------
+// // <copyright file="IDependencyResolver.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using Akka.Actor;
+
+namespace Akka.DependencyInjection
+{
+    public interface IDependencyResolver
```

Comment:
So I'm supportive of adding this abstraction, after discussing it with @to11mtm and reviewing some of Mark Seeman's advice here: https://blog.ploeh.dk/2014/05/19/di-friendly-framework/

We've already run into "conforming container" problems with Microsoft.Extensions.DependencyInjection 3.0 vs. 5.0 - so an abstraction like yours would be helpful for users who need to roll-forward before the library itself can.